x/sequencer/keeper/msg_server_update.go

```diff
@@ -56,8 +55,11 @@ func (k msgServer) UpdateOptInStatus(goCtx context.Context,
 	k.SetSequencer(ctx, seq)
 
 	// maybe set as proposer if one is needed
-	if err := k.UpdateProposerIfNeeded(ctx, seq.RollappId); err != nil {
-		return nil, errorsmod.Wrap(err, "choose proposer")
+	proposer := k.GetProposer(ctx, seq.RollappId)
+	if proposer.Sentinel() {
+		if err := k.RecoverFromHalt(ctx, seq.RollappId); err != nil {
+			return nil, err
+		}
 	}
```

Comment:
The original chooseProposer function did exactly this. It chose one if one is needed. Not sure why it's factored differently now

Reply:
for clarity and readability. by @omritoptix 's suggestion
` if one is needed.` checked by the caller